learn/getting-started/working-with-data/using-query-expressions-to-manipulate-data.md

```diff
@@ -0,0 +1,250 @@
+redirect_from:
+  - /learn/using-query-expressions-to-manipulate-data
+---
+
+This simple guide helps you understand how query expressions (a.k.a language integrated queries) can be used on data to produce new data.
+
+Ballerina supports first-class support for writing SQL-like queries to process data.
+
+Language integrated queries can process any ballerina iterable collection.
+
+In this tutorial, you will be writing queries to filter, sort and join with other data sets and produce new data sets.
+
+This tutorial includes the following steps.
+
+1. Create a new Ballerina project
+2. Define the Covid dataset to be processed
+3. Find all countries which have more than 10,000,000 Covid cases
+4. Find the top three countries by the number of reported Covid cases
+5. Join the table with an array to find the number of recovered patients of three random countries
+6. Find any discrepancies in reported Covid dataset while maintaining intermediate states
+
+## Prerequisite
+
+Following two prerequisites are needed for this tutorial. 
+
+1. Install [Ballerina Swan Lake](https://ballerina.io/downloads/)
+2. Install [VSCode](https://code.visualstudio.com/download) with [Ballerina extension](https://marketplace.visualstudio.com/items?itemName=WSO2.Ballerina)
+
+
+## Create a new Ballerina project
+
+To create a new Ballerina package, use the `bal new` command as shown below.
+
+```bash
+$ bal new query_expressions
+```
+
+This will create a new Ballerina package called `query_expressions`.
+
+The `bal new` command generates the below files inside `query_expressions`.
+
+```bash
+cd query_expressions
+tree .
+.
+├── Ballerina.toml
+└── main.bal
+    
+0 directories, 2 files
+```
+
+## Define the Covid dataset to be processed
+
+For the sake of keeping the tutorial simple, we will be using an in-memory table to store Covid dataset. Each record of type `CovidEntry` in the table, represents the Covid data related to a particular country. The `iso_code` is used to uniquely identify a country and other fields are self-explanatory
+```ballerina
+public type CovidEntry record {|
+    readonly string iso_code;
+    string country;
+    decimal cases;
+    decimal deaths;
+    decimal recovered;
+    decimal active;
+|};
+
+public final table<CovidEntry> key(iso_code) covidTable = table [
+        {iso_code: "AFG", country: "Afghanistan", cases: 159303, deaths: 7386, recovered: 146084, active: 5833},
+        {iso_code: "SL", country: "Sri Lanka", cases: 598536, deaths: 5243, recovered: 568637, active: 14656},
+        {iso_code: "US", country: "USA", cases: 69808350, deaths: 880976, recovered: 43892277, active: 25035097},
+        {iso_code: "IND", country: "India", cases: 80808350, deaths: 980976, recovered: 33892279, active: 35035095}
+    ];
+```
+Copy the above code snippet to `main.bal`. This will act as our Covid Dataset.
+
+## Find all countries which have more than 10,000,000 Covid cases
```

Comment:
```suggestion
## Find all countries which have more than 10,000,000 Covid cases
```
Need to use the `ing` form in titles. Please check all occurrences.

Reply:
Fixed